### PR TITLE
Fixed & Updated Documentation Guide and README.md

### PR DESCRIPTION
**Pull request by chetkhatri**
What were Proposed in this Pull Request ?
* README.md - Actual Installation Guide url pointed for Torch Lua
* Basic Tutorial - Installation Section Added
* README.md - Installation and Example order maintained.
* setup.sh - Removed, Hence Installation steps has given at README.md section and guideline documentation

### CODE REVIEW

**Comment by chetkhatri** [comment]
@jgehring @alexholdenmiller I think We are good now. as suggested- changed implemented.

**Comment by chetkhatri** [comment]
@alexholdenmiller Can you review and merge PR

File: docs/source/basic_tutorial.rst [hunk 1]
```diff
@@ -16,6 +16,22 @@ Its goal is to provide researchers:
...
+        git clone https://github.com/facebookresearch/ParlAI.git
+2. Install Required packages for ParlAI:
+.. code-block:: bash 
+        pip install -r requirements.txt
+3. Install ParlAI on Linux:
+.. code-block:: bash 
+        python setup.py install
+4. Several models has additional requirements:
+        a. DrQA requires installing `PyTorch <http://pytorch.org/>`_
+        b. MemNN requires installing `Lua Torch <http://torch.ch/docs/getting-started.html>`_
```
Comment (alexholdenmiller) [inline docs/source/basic_tutorial.rst]: if you click View File, the resulting rst output is still a little off.

I think you need to add more newlines: before and after the code-block lines, in particular.

File: setup.sh [hunk 2]
```diff
@@ -1,20 +0,0 @@
-#!/bin/bash
```
Comment (alexholdenmiller) [inline setup.sh]: I think I'd still like to keep this :)

**Comment by chetkhatri** [comment]
@alexholdenmiller updated

File: setup.sh (hunk 2 above)
Comment (chetkhatri) [inline setup.sh]: @alexholdenmiller Sure

File: docs/source/basic_tutorial.rst (hunk 1 above)
Comment (chetkhatri) [inline docs/source/basic_tutorial.rst]: @alexholdenmiller Added and tested by clicking View File, output seems good.

File: README.md [hunk 3]
```diff
@@ -51,6 +51,30 @@ Set of datasets to bootstrap a working dialogue model for human interaction
 - Uses zmq to talk to other toolboxes not in Python, examples of Lua Torch given.
 - Supports hogwild and batch training of models.
 
+## Requirements
```
Comment (jaseweston) [inline README.md]: not a fan of moving this

Comment (chetkhatri) [inline README.md]: @jaseweston Always Installation section would be first part instead of examples to execute on environment ready machine

Comment (jaseweston) [inline README.md]: We want to explain how easy to use it is first, and how it works.
We don't want installation at the top of the README, we want to tell people what it's about first, see e.g. https://github.com/pytorch/pytorch/blob/master/README.md

We should move installation way down the file.

Comment (chetkhatri) [inline README.md]: @jaseweston fine, sounds good.

File: docs/source/basic_tutorial.rst (hunk 1 above)
Comment (chetkhatri) [inline docs/source/basic_tutorial.rst]: @jaseweston Can you review this.

File: docs/source/basic_tutorial.rst [hunk 4]
```diff
@@ -16,6 +16,32 @@ Its goal is to provide researchers:
...
+3. Install ParlAI on Linux
+
+.. code-block:: bash 
+
+        python setup.py install
+
+4. Several models has additional requirements
+
+        a. DrQA requires installing `PyTorch <http://pytorch.org/>`_
+        b. MemNN requires installing `Lua Torch <http://torch.ch/docs/getting-started.html>`_
```
Comment (alexholdenmiller) [inline docs/source/basic_tutorial.rst]: Actually let's make this...
```rst
1. Clone ParlAI Repository:

.. code-block:: bash

        git clone https://github.com/facebookresearch/ParlAI.git ~/ParlAI

3. Install ParlAI:

.. code-block:: bash 

        cd ~/ParlAI; python setup.py develop

3. Several models has additional requirements

  a. DrQA requires installing `PyTorch <http://pytorch.org/>`

  b. MemNN requires installing `Lua Torch <http://torch.ch/docs/getting-started.html>`
```

**Comment by chetkhatri** [comment]
@alexholdenmiller Sure, I am done

**Comment by chetkhatri** [comment]
@alexholdenmiller can you review and merge it

File: README.md (hunk 3 above)
Comment (jaseweston) [inline README.md]: please revert this change then?

**Comment by chetkhatri** [comment]
@jaseweston you can merge last commit only

**Comment by alexholdenmiller** [comment]
can you remove the readme changes as jason requested so I can squash and merge automatically?

**Comment by chetkhatri** [comment]
Addressed comment @alexholdenmiller

**Comment by alexholdenmiller** [comment]
...I did end up deleting setup.sh

**Comment by chetkhatri** [comment]
@alexholdenmiller When I suggested, you said no. now you did it :) Can you merge and close this PR.

**Comment by alexholdenmiller** [comment]
I know--you were right! unfortunately it's creating a conflict, I'm not sure how to update your PR. you might need to delete setup.sh on your end and update the PR now...

**Comment by chetkhatri** [comment]
@alexholdenmiller Updated PR as suggested

**Comment by alexholdenmiller** [comment]
Thanks!
